docs/instrument-thanos.md

```diff
@@ -0,0 +1,171 @@
+COUNTER             thanos_compact_group_compactions_total                               Total number of group compaction attempts that resulted in a new block.
+COUNTER             thanos_compact_group_compaction_runs_started_total                   Total number of group compaction attempts.
+COUNTER             thanos_compact_group_compaction_runs_completed_total                 Total number of group completed compaction runs. This also includes compactor group runs that resulted with no compaction.
+COUNTER             thanos_compact_group_compactions_failures_total                      Total number of failed group compactions.
+COUNTER             thanos_compact_group_vertical_compactions_total                      Total number of group compaction attempts that resulted in a new block based on overlapping blocks.
+COUNTER             thanos_compact_garbage_collection_total                              Total number of garbage collection operations.
+COUNTER             thanos_compact_garbage_collection_failures_total                     Total number of failed garbage collection operations.
+HISTOGRAM           thanos_compact_garbage_collection_duration_seconds                   Time it took to perform garbage collection iteration.
+COUNTER             dns_lookups_total                                                    The number of DNS lookups resolutions attempts
+COUNTER             dns_failures_total                                                   The number of DNS lookup failures
+HISTOGRAM           http_request_duration_seconds                                        Tracks the latencies for HTTP requests.
+SUMMARY             http_request_size_bytes                                              Tracks the size of HTTP requests.
+COUNTER             http_requests_total                                                  Tracks the number of HTTP requests.
+SUMMARY             http_response_size_bytes                                             Tracks the size of HTTP responses.
+COUNTER             thanos_objstore_bucket_operations_total                              Total number of all attempted operations against a bucket.
+COUNTER             thanos_objstore_bucket_operation_failures_total                      Total number of operations against a bucket that failed, but were not expected to fail in certain way from caller perspective. Those errors have to be investigated.
+HISTOGRAM           thanos_objstore_bucket_operation_duration_seconds                    Duration of successful operations against the bucket
+GAUGE               thanos_objstore_bucket_last_successful_upload_time                   Second timestamp of the last successful upload to the bucket.
+GAUGE               status                                                               Represents status (0 indicates failure, 1 indicates success) of the component.
```

Comment:
This is an example of missing registry prefix. The full metric name would be `thanos_status`, but the linter can only catch part of it.